byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -26,26 +36,35 @@
 /** Servlet that handles data for stops */
 @WebServlet("/api/stop")
 public final class StopsServlet extends HttpServlet {
-  @GuardedBy("this")
-  private ArrayList<String> stops = new ArrayList<String>();
-
   private final Gson gson = new Gson();
-  // Implement datastore in routepage branch
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final Type ARRAYLIST_STRING = new TypeToken<ArrayList<String>>() {}.getType();
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
 
+  /* Passes saved destinations stops (if any) to be shown in the schedule panel */
   @Override
-  public synchronized void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(stops);
-
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> stops = Collections.emptyList();
+    try {
+      Entity entity = datastore.get(KeyFactory.createKey(Trip.KIND, 5910974510923776L));
```

Comment:
Hardcoded value is likely only valid for your local datastore. If doing temp hardcoded stup, it woudl be good to explicitly set the key of the entity in the `init` that creates the entity so that it's alway the same across all datastore instances.

Reply:
Yup I initially used the ID I created in init() (1234) in place of the Long. However createKey() uses the random-generated ID which I have not figured out how to override, so I just found took this ID value from the datastore admin page and used it.

Reply:
When you create a new entity to add to datastore, you have 3 choices for how to create the `Key` for it:

1) Create an entity with a Kind only. When you insert it into the datastore, it will get a numeric (`long`) ID value chosen automatically. To do this, you construct the entity with `new Entity(KIND)` and then when you call `datastore.put(entity)` the key will be assigned, and can be retrieved with `entity.getKey()`.
2) Choose your own value for the `long` ID. You create your Key ahead of time via the `KeyFactory`, and create the entity with it. Example: `new Entity(KeyFactory.createKey(KIND, yourChosenId))`. When you use `datastore.put(entity)`, the ID will be left with the value you chose.
3) Choose your own value for the `String` `name` of the key, and create the entity with it. Example: `new Entity(KeyFactory.createKey(KIND, yourChosenName))`. When you use `datastore.put(entity)`, the `name` will be used, and no numeric `id` will be assigned.

So if you have your 'init' setup the entities with the same key value always, you'll be creating them under that key, and can safely re-use the key elsewhere.

Reply:
Oh I see, I think I setup ID like how they told us in the walkthroughs where you make your own custom id field. I think the second method is best. I'll add my init() function again and see.